package/android/cpp/rnskia-android/RNSkOpenGLCanvasProvider.cpp

```diff
@@ -25,9 +25,30 @@ float RNSkOpenGLCanvasProvider::getScaledHeight() { return _height; }
 
 bool RNSkOpenGLCanvasProvider::renderToCanvas(
     const std::function<void(SkCanvas *)> &cb) {
-  if (_renderer != nullptr) {
-    return _renderer->run(cb, _width, _height);
+
+  if (_surfacefactory != nullptr && cb != nullptr) {
+
+    // Get the surface
+    if (_surface == nullptr) {
+      _surface = _surfacefactory->createSkSurface();
+    }
+    if (_surface) {
+
+      // If we have a render context let's make it current.
+      auto canvas = _surface->getCanvas();
+
+      // Draw into canvas using callback
+      cb(canvas);
+
+      // Commit - flush and swap buffers
+      canvas->flush();
+      return _surfacefactory->present();
```

Comment:
how do we know that the opengl context is current at this point? the flush can be moved into present via grContext->flushAndSubmit();